decode.js

```diff
@@ -10,7 +10,7 @@ module.exports = function (jwt, options) {
   if(typeof payload === 'string') {
     try {
       var obj = JSON.parse(payload);
-      if(typeof obj === 'object') {
+      if(obj && typeof obj === 'object') {
```

Comment:
Would it make more sense to check that `obj !== null` here instead? Might make it more clear that the additional condition is to guard against `typeof null => 'object'`.